### PR TITLE
add 'd' postfix to libraries built with debug build type

### PR DESCRIPTION
**Pull request by balos1**

### CODE REVIEW

**Review by gardner48 — COMMENTED** [review]
Is this change necessary? Building with `-DCMAKE_BUILD_TYPE=Debug -DCMAKE_DEBUG_POSTFIX="d"` works for me on `develop`.

File: doc/arkode/guide/source/Introduction.rst [hunk 1]
```diff
@@ -140,6 +140,9 @@ Added support for the `Ginkgo <https://ginkgo-project.github.io/>`_  linear alge
 This support includes new ``SUNMatrix`` and ``SUNLinearSolver`` implementations,
 see the sections :numref:`SUNMatrix.Ginkgo` and :numref:`SUNLinSol.Ginkgo`.
 
+Debug builds will now produce libraries with 'd' as the postfix, e.g., 
+`libsundials_cvoded.so`.
```
Comment (drreynolds) [inline doc/arkode/guide/source/Introduction.rst:144]:
```suggestion
`libsundials_arkoded.so`.
```

File: doc/cvodes/guide/source/Introduction.rst [hunk 2]
```diff
@@ -132,6 +132,9 @@ Added support for the `Ginkgo <https://ginkgo-project.github.io/>`_  linear alge
 This support includes new ``SUNMatrix`` and ``SUNLinearSolver`` implementations,
 see the sections :numref:`SUNMatrix.Ginkgo` and :numref:`SUNLinSol.Ginkgo`.
 
+Debug builds will now produce libraries with 'd' as the postfix, e.g., 
+`libsundials_cvoded.so`.
```
Comment (drreynolds) [inline doc/cvodes/guide/source/Introduction.rst:136]:
```suggestion
`libsundials_cvodesd.so`.
```

File: doc/ida/guide/source/Introduction.rst [hunk 3]
```diff
@@ -90,6 +90,9 @@ Added support for the `Ginkgo <https://ginkgo-project.github.io/>`_  linear alge
 This support includes new ``SUNMatrix`` and ``SUNLinearSolver`` implementations,
 see the sections :numref:`SUNMatrix.Ginkgo` and :numref:`SUNLinSol.Ginkgo`.
 
+Debug builds will now produce libraries with 'd' as the postfix, e.g., 
+`libsundials_cvoded.so`.
```
Comment (drreynolds) [inline doc/ida/guide/source/Introduction.rst:94]:
```suggestion
`libsundials_idad.so`.
```

File: doc/idas/guide/source/Introduction.rst [hunk 4]
```diff
@@ -104,6 +104,9 @@ Added support for the `Ginkgo <https://ginkgo-project.github.io/>`_  linear alge
 This support includes new ``SUNMatrix`` and ``SUNLinearSolver`` implementations,
 see the sections :numref:`SUNMatrix.Ginkgo` and :numref:`SUNLinSol.Ginkgo`.
 
+Debug builds will now produce libraries with 'd' as the postfix, e.g., 
+`libsundials_cvoded.so`.
```
Comment (drreynolds) [inline doc/idas/guide/source/Introduction.rst:108]:
```suggestion
`libsundials_idasd.so`.
```

File: doc/kinsol/guide/source/Introduction.rst [hunk 5]
```diff
@@ -106,6 +106,9 @@ Added support for the `Ginkgo <https://ginkgo-project.github.io/>`_  linear alge
 This support includes new ``SUNMatrix`` and ``SUNLinearSolver`` implementations,
 see the sections :numref:`SUNMatrix.Ginkgo` and :numref:`SUNLinSol.Ginkgo`.
 
+Debug builds will now produce libraries with 'd' as the postfix, e.g., 
+`libsundials_cvoded.so`.
```
Comment (drreynolds) [inline doc/kinsol/guide/source/Introduction.rst:110]:
```suggestion
`libsundials_kinsold.so`.
```

**Review by drreynolds — COMMENTED** [review]
I've added some suggestions to this PR if we want to go forward with it.  However, I do wonder whether it is even necessary at all, since this user can just add one CMake option to get the same behavior, whereas if we approve this PR then that *requires* all future users to have debug libraries with new names.

**Comment by balos1** [comment]
> Is this change necessary? Building with `-DCMAKE_BUILD_TYPE=Debug -DCMAKE_DEBUG_POSTFIX="d"` works for me on `develop`.

I did not realize that, ill check because the original issue was with using sundials as a subproject.

**Comment by balos1** [comment]
This does not seem to be needed, so I am closing.

**Pull request closed without merging**